src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -14,85 +14,133 @@
 namespace CloudNative.CloudEvents.NewtonsoftJson
 {
     /// <summary>
-    /// Formatter that implements the JSON Event Format
+    /// Formatter that implements the JSON Event Format.
     /// </summary>
-    public class JsonEventFormatter : ICloudEventFormatter
+    /// <remarks>
+    /// <para>
+    /// When encoding CloudEvent data, the behavior depends on the data content type of the CloudEvent
+    /// and the type of the <see cref="CloudEvent.Data"/> property value, following the rules below.
+    /// </para>
+    /// <list type="bullet">
+    /// <item><description>
+    /// If the data value is null, the content is empty for a binary mode message, and neither the "data"
+    /// nor "data_base64" property is populated in a structured mode message.
+    /// </description></item>
+    /// <item><description>
+    /// If the data content type is absent or has a media type of "application/json", the data is encoded as JSON.
+    /// If the data is already a <see cref="JToken"/>, that is serialized directly as JSON. Otherwise, the data
+    /// is converted using the <see cref="JsonSerializer"/> passed into the constructor, or a
+    /// default serializer.
+    /// </description></item>
+    /// <item><description>
+    /// Otherwise, if the data content type has a media type beginning with "text/" and the data value is a string,
```

Comment:
What about having an explicit parameter that can be set by the user to control how the data should be serialized?

Reply:
Not sure... at that point it feels like the user's doing the event formatter's job for it. They could always change the value to be a JToken and set the content type to application/json if they want it to be JSON. Let's talk more though - I'm not set on what we've got now.